lib/dry/schema/processor.rb

```diff
@@ -84,7 +84,7 @@ def new(options = nil, &block)
       #
       # @api public
       def call(input)
-        Result.new(input, input: input, message_compiler: message_compiler) do |result|
+        Result.new(input.dup, message_compiler: message_compiler) do |result|
```

Comment:
Previously we were mutating only inner hashes of input, now we're using `Hash#replace` as a `Dry::Schema::Result#replace`, so non-frozen input is needed.